code/game/objects/items/mixing_bowl.dm

```diff
@@ -0,0 +1,159 @@
+
+/obj/item/mixing_bowl
+	name = "mixing bowl"
+	desc = "Mixing it up in the kitchen."
+	flags = OPENCONTAINER
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "mixing_bowl"
+	var/max_n_of_items = 25
+	var/dirty = FALSE
+	var/clean_icon = "mixing_bowl"
+	var/dirty_icon = "mixing_bowl_dirty"
+
+/obj/item/mixing_bowl/New()
+	..()
+	create_reagents(100)
+
+/obj/item/mixing_bowl/attackby(obj/item/I, mob/user, params)
+	if(dirty)
+		if(istype(I, /obj/item/weapon/soap))
+			user.visible_message("<span class='notice'>[user] starts to scrub [src].</span>", "<span class='notice'>You start to scrub [src].</span>")
+			if(do_after(user, 20 * I.toolspeed, target = src))
+				clean()
+				user.visible_message("<span class='notice'>[user] has scrubbed [src] clean.</span>", "<span class='notice'>You have scrubbed [src] clean.</span>")
+			return 1
+		else
+			to_chat(user, "<span class='warning'>You should clean [src] before you use it for food prep.</span>")
+		return 0
+	if(is_type_in_list(I, cooking_ingredients[RECIPE_MICROWAVE]) || is_type_in_list(I, cooking_ingredients[RECIPE_GRILL]) || is_type_in_list(I, cooking_ingredients[RECIPE_OVEN]) || is_type_in_list(I, cooking_ingredients[RECIPE_CANDY]))
+		if(contents.len>=max_n_of_items)
+			to_chat(user, "<span class='alert'>This [src] is full of ingredients, you cannot put more.</span>")
+			return 1
+		if(istype(I, /obj/item/stack) && I:amount > 1)
+			new I.type (src)
+			I:use(1)
+			user.visible_message("<span class='notice'>[user] has added one of [I] to [src].</span>", "<span class='notice'>You add one of [I] to [src].</span>")
+		else
+			if(!user.drop_item())
+				to_chat(user, "<span class='notice'>\The [I] is stuck to your hand, you cannot put it in [src]</span>")
+				return 0
+			I.forceMove(src)
+			user.visible_message("<span class='notice'>[user] has added [I] to [src].</span>", "<span class='notice'>You add [I] to [src].</span>")
+	else if(is_type_in_list(I, list(/obj/item/weapon/reagent_containers/glass, /obj/item/weapon/reagent_containers/food/drinks, /obj/item/weapon/reagent_containers/food/condiment)))
+		if(!I.reagents)
+			return 1
+		for(var/datum/reagent/R in I.reagents.reagent_list)
+			if(!(R.id in cooking_reagents[RECIPE_MICROWAVE]) && !(R.id in cooking_reagents[RECIPE_GRILL]) && !(R.id in cooking_reagents[RECIPE_OVEN]) && !(R.id in cooking_reagents[RECIPE_CANDY]))
+				to_chat(user, "<span class='alert'>Your [I] contains components unsuitable for cookery.</span>")
+				return 1
+	else
+		to_chat(user, "<span class='alert'>You have no idea what you can cook with  [I].</span>")
+		return 1
+
+/obj/item/mixing_bowl/attack_self(mob/user)
+	var/dat = ""
+	if(dirty)
+		dat = {"<TT>This [src] is dirty!<BR>Please clean it before use!</TT>"}
```

Comment:
I haven't seen `<tt>` used in years.

Reply:
I would do away with it personally as it's depreciated nowadays, but it's really down to the maintainers as I haven't really seen any standards referred to with our HTML. I would just replace it with a `<font>` or `<span>` personally.